src/web/client/extension.ts

```diff
@@ -268,12 +269,95 @@ export function processWillSaveDocument(context: vscode.ExtensionContext) {
+        createWebWorkerInstance(context);
     }
 }
 
+let worker: Worker;
+
+export function processOnTabChange(context: vscode.ExtensionContext) {
+    try {
+        context.subscriptions.push(
+            vscode.window.tabGroups.onDidChangeTabs((event) => {
+                event.opened.concat(event.changed).forEach(tab => {
+                    if (tab.input instanceof vscode.TabInputCustom || tab.input instanceof vscode.TabInputText) {
+                        const document = tab.input;
+                        const entityId = getFileEntityId(document.uri.fsPath);
+                        if (entityId) {
+                            worker.postMessage({
+                                afrConfig: {
+                                    swpId: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.SHAREWORKSPACE_ID
+                                    ) as string,
+                                    swptenantId: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.TENANT_ID
+                                    ) as string,
+                                    discoveryendpoint: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.DISCOVERY_ENDPOINT
+                                    ) as string,
+                                    swpAccessToken: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.ACCESS_TOKEN
+                                    ) as string,
+                                },
+                                entityId,
+                            });
+                        }
+                    }
+                });
+            })
+        );
+    } catch (error) {
+        // TODO: add Telemetry
+    }
+}
+
+export function createWebWorkerInstance(
+    context: vscode.ExtensionContext
+) {
+    const webworkerMain = vscode.Uri.joinPath(
+        context.extensionUri,
+        "dist/web/webworker.worker.js"
+    );
+
+    const workerUrl = new URL(webworkerMain.toString());
+
+    fetch(workerUrl)
+        .then((response) => response.text())
+        .then((workerScript) => {
+            const workerBlob = new Blob([workerScript], {
+                type: "application/javascript",
+            });
+
+            const workerUrl = URL.createObjectURL(workerBlob);
+
+            worker = new Worker(workerUrl);
+
+            worker.onmessage = (event) => {
+                const { data } = event;
+
+                WebExtensionContext.containerId = event.data.containerId;
+
+                if (data.type === "member-removed") {
+                    WebExtensionContext.removeConnectedUserInContext(
+                        data.userId
+                    );
+                }
+                if (data.type === "client-data") {
```

Comment:
lets use a const for events like "client-data"

Reply:
Done: https://github.com/microsoft/powerplatform-vscode/pull/745/commits/1fcfe2e2841eb58e74681c58dc234b5dc6d411e9